### PR TITLE
scheduler with Duplicated ReplicaSchedulingStrategy

### PR DESCRIPTION
**Pull request by qianjun1993**
Signed-off-by: junqian <junqian@tencent.com>

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Part of #344

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
NONE

```release-note

```

### CODE REVIEW

File: pkg/scheduler/core/generic_scheduler.go [hunk 1]
```diff
@@ -190,3 +191,24 @@ func (g *genericScheduler) chooseSpreadGroup(spreadGroup *util.SpreadGroup) []st
...
 }
+
+func (g *genericScheduler) assignReplicas(clusters []*clusterv1alpha1.Cluster, replicaSchedulingStrategy *policyv1alpha1.ReplicaSchedulingStrategy, object *workv1alpha1.ObjectReference) ([]workv1alpha1.TargetCluster, error) {
+	if len(clusters) == 0 {
+		return nil, fmt.Errorf("no clusters available to schedule")
+	}
+	targetClusters := make([]workv1alpha1.TargetCluster, len(clusters))
+
+	if object.Replicas > 0 {
+		if replicaSchedulingStrategy.ReplicaSchedulingType == policyv1alpha1.ReplicaSchedulingTypeDuplicated {
```
Comment (mrlihanbo) [inline pkg/scheduler/core/generic_scheduler.go:202]: It seems like we need to check if replicaSchedulingStrategy == nil first?

Comment (qianjun1993) [reply pkg/scheduler/core/generic_scheduler.go:202]: `if object.Replicas > 0 && replicaSchedulingStrategy != nil { ` 
like this ?

Comment (mrlihanbo) [reply pkg/scheduler/core/generic_scheduler.go:202]: I think it is ok.  I will run it locally.

File: pkg/scheduler/core/generic_scheduler.go [hunk 2]
```diff
@@ -190,3 +191,24 @@ func (g *genericScheduler) chooseSpreadGroup(spreadGroup *util.SpreadGroup) []st
...
+			for i, cluster := range clusters {
+				targetClusters[i] = workv1alpha1.TargetCluster{Name: cluster.Name, Replicas: object.Replicas}
+			}
+			return targetClusters, nil
+		}
+	}
+
+	for i, cluster := range clusters {
+		targetClusters[i] = workv1alpha1.TargetCluster{Name: cluster.Name}
+	}
```
Comment (RainbowMango) [inline pkg/scheduler/core/generic_scheduler.go:212]: Does this always make a copy of `clusters`? for non-replica resources?

Comment (qianjun1993) [reply pkg/scheduler/core/generic_scheduler.go:212]: Yes

**Pull request merged**